common/rpc/interceptor/telemetry.go

```diff
@@ -364,10 +380,47 @@ func (ti *TelemetryInterceptor) handleError(
 			}
 		}
 		metrics.ServiceFailures.With(metricsHandler).Record(1)
+		logTags = append(logTags, ti.getWorkflowTags(req)...)
 		ti.logger.Error("service failures", append(logTags, tag.Error(err))...)
 	}
 }
 
+func (ti *TelemetryInterceptor) getWorkflowTags(
+	req interface{},
+) []tag.Tag {
+	if executionGetter, ok := req.(ExecutionGetter); ok {
+		execution := executionGetter.GetExecution()
+		return []tag.Tag{tag.WorkflowID(execution.WorkflowId), tag.WorkflowRunID(execution.RunId)}
+	}
+	if workflowExecutionGetter, ok := req.(WorkflowExecutionGetter); ok {
+		execution := workflowExecutionGetter.GetWorkflowExecution()
+		return []tag.Tag{tag.WorkflowID(execution.WorkflowId), tag.WorkflowRunID(execution.RunId)}
+	}
+	if taskTokenGetter, ok := req.(TaskTokenGetter); ok {
+		taskTokenBytes := taskTokenGetter.GetTaskToken()
+		if len(taskTokenBytes) == 0 {
+			return []tag.Tag{}
+		}
+		// Special case for avoiding deprecated RespondQueryTaskCompleted API token which does not have workflow id.
+		if _, ok := req.(*workflowservice.RespondQueryTaskCompletedRequest); !ok {
+			taskToken, err := ti.serializer.Deserialize(taskTokenBytes)
+			if err != nil {
+				ti.logger.Error("unable to deserialize task token", tag.Error(err))
+				return []tag.Tag{}
+			}
+			return []tag.Tag{tag.WorkflowID(taskToken.WorkflowId), tag.WorkflowRunID(taskToken.RunId)}
+		}
```

Comment:
nit: I think it'd be clearer if it was

```suggestion
		// Special case for avoiding deprecated RespondQueryTaskCompleted API token which does not have workflow id.
		if _, ok := req.(*workflowservice.RespondQueryTaskCompletedRequest); ok {
		        return []tag.Tag{}
		}
		taskToken, err := ti.serializer.Deserialize(taskTokenBytes)
		if err != nil {
			ti.logger.Error("unable to deserialize task token", tag.Error(err))
			return []tag.Tag{}
		}
		return []tag.Tag{tag.WorkflowID(taskToken.WorkflowId), tag.WorkflowRunID(taskToken.RunId)}
```

Reply:
Yeah it is clearer this way.. Thanks @stephanos!